src/vs/platform/window/common/window.ts

```diff
@@ -128,17 +128,57 @@ export function isFileToOpen(uriToOpen: IWindowOpenable): uriToOpen is IFileToOp
 }
 
 export const enum MenuSettings {
+	MenuStyle = 'window.menuStyle',
 	MenuBarVisibility = 'window.menuBarVisibility'
 }
 
+export const enum MenuStyleConfiguration {
+	CUSTOM = 'custom',
+	NATIVE = 'native',
+	INHERIT = 'inherit',
+}
+
+export function hasNativeContextMenu(configurationService: IConfigurationService, titleBarStyle?: TitlebarStyle): boolean {
+	if (isWeb) {
+		return false;
+	}
+
+	const nativeTitle = hasNativeTitlebar(configurationService, titleBarStyle);
+	const windowConfigurations = configurationService.getValue<IWindowSettings | undefined>('window');
+
+	if (windowConfigurations?.menuStyle === MenuStyleConfiguration.NATIVE) {
+		// Do not support native menu with custom title bar
+		if (!isMacintosh && !nativeTitle) {
+			return false;
+		}
```

Comment:
let me ask, why? many linux desktop environments support global menu for example which will work with custom title bar